torchgeo/datasets/spacenet1.py

```diff
@@ -0,0 +1,215 @@
+"""Spacenet1 dataset."""
```

Comment:
I wonder if we should collect all spacenet datasets into a single `spacenet.py` file. Do you know how much these datasets have in common?

Reply:
The datasets don't really have much in common but that doesn't necessarily preclude us from making a single Spacenet meta class.  

Briefly,

Spacenet 1 & Spacenet 2 - Building detection over optical imagery
Spacenet 3 - Road segmentation
Spacenet 4 - Off nadir building detection
Spacenet 5 - Road segmentation and optimal route classification
Spacenet 6 - SAR and EO Building Detection
Spacenet 7 -  Building Detection Time Series

From an implementation perspective, it always follows this formula: **1 or more GeoTIFFs and 1 GeoJSON**.

Reply:
Yeah, let's put them all in `spacenet.py` then. They're the same family of datasets even if they don't share the same structure.

Reply:
Latest changes:
* Create single spacenet.py for all future spacenet datasets
* Create single spacenet directory for all spacenet test data
* Create single test_spacenet.py for testing all spacenet datasets